File: problems/problems_1018/Solution.java
package problems.problems_1018;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] nums = jsonArrayToIntArray(inputJsonValues[0]);
        return JSON.toJSON(prefixesDivBy5(nums));
    }
}

File: problems/problems_1018/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.prefixesDivBy5(test_input)

    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        n = len(nums)
        ans = [False] * n
        cur = 0
        for i in range(n):
            cur = (cur * 2 + nums[i]) % 5
            ans[i] = cur == 0
        return ans
